### PR TITLE
Remove scrna

### PR DESCRIPTION
**Pull request by ben-heil**
This is a pretty short PR, it just adds a script that removes scRNA samples from the compendium.

Recount had two methods of labeling scRNAseq. I though their pattern-based one (below) was more believable. However, it seems to call a lot of very sparse RNA-seq in the upper ranges, so I also set a manual 75% sparsity cutoff. 
image

### CODE REVIEW

File: renv.lock [hunk 1]
```diff
@@ -0,0 +1,682 @@
+{
```
Comment (jjc2718) [inline renv.lock:1]: Do you like `renv` better than using conda with R packages? I've always found conda impossible to use with R - interested in trying something different next time I have to write some R code.

File: saged/remove_scrnaseq.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+            # Skip malformed lines
+            except TypeError:
+                continue
+
+            try:
+                if sparsity > .75 or recount_pred == 'scrna-seq':
+                    continue
+                else:
+                    out_file.write(line)
+            except ValueError as e:
```
Comment (jjc2718) [inline saged/remove_scrnaseq.py:79]: Why would a `ValueError` happen here? May be worth adding a comment.

File: saged/remove_scrnaseq.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    # Drop rows without sample ids and duplicate rows
+    metadata = metadata[metadata['external_id'].notna()]
+    metadata = metadata.drop_duplicates(subset=['external_id', 'study'])
+    # Set the index to the sample id for faster access
+    metadata = metadata.set_index('external_id')
+
+    LINES_IN_FILE = 317259
+
+    with open(args.count_file, 'r') as count_file:
+        out_file = open(args.out_file, 'w')
```
Comment (jjc2718) [inline saged/remove_scrnaseq.py:29]: Do you want to close this file handle somewhere?

File: saged/remove_scrnaseq.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+            except ValueError:
+                continue
+
+            # This works for int zeros and float zeros
+            zero_count = counts.count(0)
+            sparsity = zero_count / total_genes
+            try:
+                sample_metadata = metadata.loc[sample, :]
+            except KeyError:
+                # print(e)
```
Comment (jjc2718) [inline saged/remove_scrnaseq.py]: Can you get rid of this line? I'm guessing it was for debugging.

**Review by jjc2718 — APPROVED** [review]
Looks good! Just a few small comments.

File: renv.lock (hunk 1 above)
Comment (ben-heil) [reply renv.lock:1]: Yeah, it's great! (Or at least it's great so far, I haven't tried to load an renv environment on another machine so we'll see). On the download side it mostly just works like I want Conda to e.g. it automatically detects the difference between bioconductor and CRAN packages

File: saged/remove_scrnaseq.py (hunk 2 above)
Comment (ben-heil) [reply saged/remove_scrnaseq.py:79]: ValueErrors crop up when a line is malformed but still passes through `len`. Went ahead and added a comment

File: saged/remove_scrnaseq.py (hunk 3 above)
Comment (ben-heil) [reply saged/remove_scrnaseq.py:29]: Good call

File: saged/remove_scrnaseq.py (hunk 4 above)
Comment (ben-heil) [reply saged/remove_scrnaseq.py]: Right, good catch!

**Pull request merged**